src/codegen/generators/llvm.rs

```diff
@@ -221,6 +221,19 @@ impl<'a> Llvm<'a> {
         }
     }
 
+    /// create a null pointer
+    pub fn create_null_ptr(&self) -> Result<TypeAndValue<'a>, CompileError> {
```

Comment:
Review: I don't know how much I like this. I have no type information at this position, so I assumed i32, which does not matter since it's a null pointer.
I also returned type information on the fly, but it might make more sense to have a NULL type in the index.